plugins/woocommerce/src/Blocks/BlockTypesController.php

```diff
@@ -67,6 +75,31 @@ function ( $ret ) {
 		);
 	}
 
+	/**
+	 * Cache registered blocks that have WooCommerce blocks as their parents. Adds the value to the
+	 * `registered_blocks_with_woocommerce_parents` class property.
+	 *
+	 * @return void
+	 */
+	public function cache_registered_blocks_with_woocommerce_parent() {
```

Comment:
Since we're unsure if `init` is too early, could we lazy load this? 

Create a function named `get_registered_blocks_with_woocommerce_parents()` and then cache if `null` and `init` has ran using [did_action](https://developer.wordpress.org/reference/functions/did_action/).

Reply:
This also avoids caching if nothing needs this cache.

Reply:
Thanks for the suggestion! Nice idea. I implemented this.